pkg/services/store/entity/sqlstash/folder_support.go

```diff
@@ -54,10 +54,10 @@ func updateFolderTree(ctx context.Context, tx *session.SessionTx, tenant int64)
 		}
 		all = append(all, &folder)
 	}
-	err = rows.Close()
-	if err != nil {
-		return err
-	}
+
+	defer func() {
+		err = rows.Close()
+	}()
```

Comment:
I think the linter is misleading you here, I had to think a bit about this before I figured out what is really going on here: this ends up being the equivalent to not checking the error, see https://go.dev/play/p/WJdVVDR91K- for a small test re: the behavior of overriding the err variable in a deferral (that's scenario B, this change is going from scenario A->B, for that to work you'd have to progress to named return values in C [which I'm not sure I'm a big fan of]).

And since the defer comes after the for-loop where there is a chance for `rows` to have been successfully initialized and `rows.Scan` to fail and return an error on line 53, which is what I think the linter detects and complain about, but seemingly cannot tell is the case anymore after you've added the defer.

There is also another risk with the defer, if there's an error on line 64, 69, or 75 that'll be overridden by whatever result comes out of `rows.Close`, including nil, see https://go.dev/play/p/w3J64xQ75da .

Reply:
(I’m looking into fixing this 😊)

Reply:
good catch and thank you for the example, I had the impression that it would return the error if it was assigned inside defer.

I don't like this `sqlclosecheck` linter check either and also it enforces using defer also to non SQL `Close()` (therefore I was tempted to temporarily disable it).

Reply:
:eyes: I'm quickly looking at the source of the linter, and imo this seems like a harmful linter. I'd suggest permanently disabling it, it doesn't detect improper use of defer, and it doesn't detect when all return paths are covered and a defer is unnecessary, so I'd say it encourages worse coding practices and introduces more bugs than it's likely to ever catch.

The counterpoint to that is that we really want Close to be called so we don't leak SQL connections, but again, this doesn't actually check that, so it sort of coaxes the user into writing hard-to-parse code that passes the linter rather than detecting actual problems.